### PR TITLE
:recycle: Implement fixed 2d array

### PR DESCRIPTION
**Pull request by ClementBaziret**
This pull request changes the way the data is stored.

It was originally stored in a : `Vec<Status>`

but is now stored in an array with fixed size : `[[CellType; 20]; 20]`

This will allow better performance in the future.

### CODE REVIEW

File: src/my_board.rs [hunk 1]
```diff
@@ -17,71 +17,48 @@ impl Status {
...
-        }
+        self.board = [[CellType::Empty; 20]; 20];
     }
 
-    pub fn calculate_next_move(&self) -> (usize, usize) {
+    pub fn calculate_next_move(&self) -> (u8, u8) {
         let root = self.generate_tree();
         let mut best_move = &Move {
-            move_position: usize::MAX,
+            // I don't really know what values to put there, could you help me ?
```
Comment (C0Florent) [inline src/my_board.rs:59]: We'll think about that when implementing a proper decision tree

**Review by C0Florent — APPROVED** [review]
This seems good to me!

File: src/my_board.rs (hunk 1 above)
Comment (ClementBaziret) [reply src/my_board.rs:59]: Okay 🫡

**Review by C0Florent — APPROVED** [review]

**Review by Phuitric — APPROVED** [review]
That's good

**Pull request merged**